ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,518 @@
+            /// frontend itself, not a framework. For example, all output ports without consumers
+            /// may be considered as outputs.
+            ///
+            /// \return A vector of output place references
+            virtual std::vector<Place::Ptr> get_outputs() const;
+
+            /// \brief Returns a tensor place by a tensor name following framework conventions, or
+            /// nullptr if a tensor with this name doesn't exist.
+            /// \param tensorName Name of tensor
+            /// \return Tensor place corresponding to specifed tensor name
+            virtual Place::Ptr get_place_by_tensor_name(const std::string& tensorName) const;
+
+            /// \brief Returns an operation place by an operation name following framework
+            /// conventions, or nullptr if an operation with this name doesn't exist. \param
+            /// operationName Name of operation \return Place representing operation
+            virtual Place::Ptr get_place_by_operation_name(const std::string& operationName);
+
+            /// \brief Returns an input port place by operation name and appropriate port index
+            /// \param operationName Name of operation
+            /// \param outputPortIndex Index of input port for this operation
+            /// \return Place representing input port of operation
+            virtual Place::Ptr
+                get_place_by_operation_and_input_port(const std::string& operationName,
+                                                      int inputPortIndex);
+
+            /// \brief Returns an output port place by operation name and appropriate port index
+            /// \param operationNameNname of operation
+            /// \param outputPortIndex Index of output port for this operation
+            /// \return Place representing output port of operation
+            virtual Place::Ptr
+                get_place_by_operation_and_output_port(const std::string& operationName,
+                                                       int outputPortIndex);
+
+            ///// Naming and annotation  /////
+
+            /// \brief Sets name for tensor. Overwrites existing names of this place
+            /// \param operation Tensor place
+            /// \param newName New name for this tensor
+            virtual void set_name_for_tensor(Place::Ptr tensor, const std::string& newName);
+
+            /// \brief Adds new name for tensor
+            /// \param operation Tensor place
+            /// \param newName New name to be added to this place
+            virtual void add_name_for_tensor(Place::Ptr tensor, const std::string& newName);
+
+            /// \brief Sets name for operation. Overwrites existing names of this place
+            /// \param operation Operation place
+            /// \param newName New name for this operation
+            virtual void set_name_for_operation(Place::Ptr operation, const std::string& newName);
```

Comment:
I think we should have `add_name_for_operation`
```

Reply:
My personal preference is to not add new methods 'just for consistency' which will be 'not implemented' and not planned to be implemented by any frontend and not be used by any client. But if we have clear use case for this, I'm ok to add this to interface now. @slyalin what is your opinion?

Reply:
@GlebKazantaev, that's a good next step if we allow it in ngraph. Currently it cannot be implemented as we don't support multiple names for an operation in ngraph (am I right?). Let keep it as a leftover (@nosovmik, please create a ticket for future improvements).

Reply:
Separate ticket #56772 is created for this